src/main/kotlin/com/kllawesome/api/App.kt

```diff
@@ -0,0 +1,38 @@
+package com.kllawesome.api
+
+import com.kllawesome.api.util.coroutineHandler
+import io.vertx.core.http.HttpServer
+import io.vertx.ext.web.Router
+import io.vertx.ext.web.RoutingContext
+import io.vertx.ext.web.handler.BodyHandler
+import io.vertx.kotlin.coroutines.CoroutineVerticle
+import io.vertx.kotlin.coroutines.awaitResult
+
+
+class App : CoroutineVerticle() {
+
+    suspend override fun start() {
+        // Build Vert.x Web router
+        val router = Router.router(vertx)
+        router.route().handler(BodyHandler.create())
+        router.post("/api/upload").coroutineHandler { ctx -> uploadConfig(ctx) }
```

Comment:
We should use something to generate API documentation for us so it's not all in the code. Not sure what works best with this language/framework but I've always liked [apiDoc](http://apidocjs.com/) myself.

Reply:
Looks good to me. I usually write my Swagger by hand but this tool looks nice.

Reply:
Ah right Swagger! I haven't had a chance to use it but I've heard good things. I'm fine with either myself! Ideally I think we should use Github Pages to host the API docs for this repo myself